test/unit/tech/html5.test.js

```diff
@@ -442,3 +444,30 @@ test('Html5#reset calls Html5.resetMediaElement when called', function() {
 
   Html5.resetMediaElement = oldResetMedia;
 });
+
+test('Html5#setSrc clears currentSource_ after loadstart', function() {
+
+  let disposed = false;
+  let thing = {
+    off: () => {},
+    one: (el, type, fun) => {
+      el.one(type, Fn.bind(thing, fun));
+    },
```

Comment:
I'd create an object that inherited from EventTarget right in the test here and use that.